packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -97,8 +96,24 @@ public ClientProvider(InputClient inputClient)
                 ? [PipelineProperty.AsParameter, _apiKeyAuthField.AsParameter, _endpointParameter]
                 : [PipelineProperty.AsParameter, _endpointParameter];
 
+            foreach (InputParameter p in inputClient.Parameters)
+            {
+                var parameterProvider = ClientModelPlugin.Instance.TypeFactory.CreateParameter(p);
+                if (!p.IsEndpoint && parameterProvider != PipelineProperty.AsParameter && parameterProvider != _apiKeyAuthField?.AsParameter)
+                {
+                    _subClientInternalConstructorParams.Add(parameterProvider);
+                    FieldProvider field = new(
```

Comment:
Does it matter that this FieldProvider is a different instance than the one returned in BuildFields?